menu_delivery/p2pnode.py

```diff
@@ -0,0 +1,136 @@
+from p2pnetwork.node import Node
+from rich.live import Live
+from rich.table import Table
+import time
+import socket
+
+
+class P2PNode(Node):
+    """P2P Node Module extended from p2pnetwork.nodem with a few changes
+    to the run function where it updates the server visuals every loop
+
+    Args:
+        Node (Node): extends the node class from p2pnetwork
+    """
+
+    def __init__(self, host, port, id=None, callback=None, max_connections=0):
+        """initializes the Node for connecting from/to.
+        generates the Table for the server's output and the message sent.
+
+        Args:
+            host (string): host IP
+            port (string): host Port
+            id (string, optional): host id. Defaults to None.
+            callback (function, optional): callback function. Defaults to None.
+            max_connections (int, optional): number of max connections. Defaults to 0.
+        """
+        super(P2PNode, self).__init__(host, port, id, callback, max_connections)
+        self.msg = None
+        self.server_table = Table()
+
+    def node_message(self, node, data):
+        """Determines how the messages sent by the nodes will be formatted
+
+        Args:
+            node (Node): _description_
+            data (string): _description_
+        """
+        self.msg = (self.id, node.id, data)
+
+    def run(self):
+        """The main loop of the thread that deals with connections from other nodes on the network. When a
+        node is connected it will exchange the node id's. First we receive the id of the connected node
+        and secondly we will send our node id to the connected node. When connected the method
+        inbound_node_connected is invoked."""
+
+        with Live(self.server_table, refresh_per_second=4):
+            prior_order = ""
+            while (
+                not self.terminate_flag.is_set()
+            ):  # Check whether the thread needs to be closed
+                try:
+                    time.sleep(0.4)
+                    if self.msg != None:
+                        order_number = self.msg[2]["order_number"]
+                        if prior_order != order_number:
+                            order_description = {
+                                k: self.msg[2][k]
```

Comment:
u should change the variable k here to something else (its used twice for no good reason) as well as id say u should split this up into multiple lines because this is really dense lol